patches/tModLoader/Terraria/Social/Steam/WorkshopHelper.cs.patch

```diff
@@ -160,7 +162,43 @@
  						CreateItem();
  					}
  				}
-@@ -240,10 +_,10 @@
+@@ -205,6 +_,8 @@
+ 				}
+ 
+ 				private void CreateItem() {
++					Logging.tML.Info("Creating Item " + _entryData.Title);
++
+ 					CoreSocialModule.SetSkipPulsing(shouldSkipPausing: true);
+ 					SteamAPICall_t hAPICall = SteamUGC.CreateItem(SteamUtils.GetAppID(), EWorkshopFileType.k_EWorkshopFileTypeFirst);
+ 					_createItemHook.Set(hAPICall, CreateItemResult);
+@@ -212,11 +_,13 @@
+ 				}
+ 
+ 				private void CreateItemResult(CreateItemResult_t param, bool bIOFailure) {
++					// This call can lead to creating a bunch of empty items due to failing to write the Manifest.json contents. Removing it here minimizes duplicates on rapid/repeated errored publishing - Solxan
+-					if (param.m_bUserNeedsToAcceptWorkshopLegalAgreement) {
++					/* if (param.m_bUserNeedsToAcceptWorkshopLegalAgreement) {
+ 						_issueReporter.ReportDelayedUploadProblem("Workshop.ReportIssue_FailedToPublish_UserDidNotAcceptWorkshopTermsOfService");
+ 						_endAction(this);
+ 					}
++					else */
+-					else if (param.m_eResult == EResult.k_EResultOK) {
++					if (param.m_eResult == EResult.k_EResultOK) {
+ 						_publishedFileID = param.m_nPublishedFileId;
+ 						UpdateItem();
+ 					}
+@@ -233,17 +_,19 @@
+ 				private void UpdateItem() {
+ 					string headerText = GetHeaderText();
+ 					if (!TryWritingManifestToFolder(_entryData.ContentFolderPath, headerText)) {
+-						_endAction(this);
+-						return;
++						// If this fails to write, any future attempts to publish the item will always create a new item. We don't want that, so give clear message that something is wrong. - Solxan
++						Logging.tML.Fatal("Failed to write critical publishing information to the item content folder. Ending user interactions with the Workshop immediately to preserve integrity.");
```

Comment:
Force ending the game instance is an excessive solution to #1706  which prevents tossing an exception.
The main comment remains the goal, but reviews say don't like current. Is there a good alternative?

Possibly consider switching to write to a static field the item publish info (publish ID, content folder source), and have any future attempts be blocked by the existence of real data in the field and display an appropriate error result prompt to inform why publishing ingame is blocked. Two main concerns with this: is convoluted and still doesn't stop someone from just restarting game to try again.

Reply:
People will definitely restart to try again. Is there nothing that prevents publishing a workshop item with the same name as one that already exists? Particularly with the same author? We could require they delete the old one? (Haven't followed most of the discussion around workshop so please excuse the question if it's dumb due to lacking context)